tests/foreman/cli/test_role.py

```diff
@@ -123,3 +123,189 @@ def test_positive_update_name(self):
+
+        @caseautomation: notautomated
+        """
+
+    @tier1
+    def test_negative_override_non_overridable_filter(self):
+        """Override non overridable filter
+
+        @id: 163313eb-4401-4bb0-bf9a-58030251643b
+
+        @steps:
+        1. Attempt to override a filter to which taxonomies cannot be
+        associated.
+        e.g Architecture filter
+
+        @assert: Filter is not overrided as taxonomies cannot be applied to
+        that filter
+
+        @caseautomation: notautomated
+        """
+
+    @tier1
+    def test_positive_create_overridable_filter(self):
+        """Create overridable filter in role
+
+        @id: 47816636-d215-45a8-9d21-495b1e193913
+
+        @steps:
+        1. Create a filter to which taxonomies can be associated.
+        e.g Domain filter
+        2. Override a filter with some taxonomies
+
+        @assert:
+        1. Filter is created with taxonomies
+        2. Override check is set to true
+        3. Filter doesnt inherits taxonomies from role
+
+        @caseautomation: notautomated
+        """
+
+    @tier1
+    def test_positive_update_role_taxonomies(self):
+        """Update role taxonomies which applies to its non-overrided filters
+
+        @id: 988cf8c6-8f6e-49de-be54-d17085f260b6
+
+        @steps:
+        1. Update existing role with different taxonomies
+
+        @assert: The taxonomies are applied only to non-overrided role filters
+
+        @caseautomation: notautomated
+        """
+
+    @tier1
+    def test_negative_update_role_taxonomies(self):
+        """Update role taxonomies which doesnt applies to its overrided filters
+
+        @id: 9d0d94fa-34ee-41a0-868d-4dc7d774fb02
+
+        @steps:
+        1. Update existing role with different taxonomies
+
+        @assert: The overridden role filters are not updated
+
+        @caseautomation: notautomated
+        """
+
+    @tier1
+    def test_positive_disable_filter_override(self):
+        """Unset override resets filter taxonomies
+
+        @id: f00f4f1f-caee-47ac-b02e-7802300d08a8
+
+        @steps:
+        1. Create role with overridden filter having different taxonomies than
+        its role.
+        2. Unset the override flag in above role filter
+
+        @assert: The taxonomies of filters resets/synced to role taxonomies
+
+        @caseautomation: notautomated
+        """
+
+    @tier1
+    def test_positive_access_resources_from_role_taxonomies(self):
+        """Test user access resources from taxonomies of assigned role
+
+        @id: 754cf23e-43ed-4d2b-b4ec-dacc77bc009d
+
+        @steps:
+        1. Create role with taxonomies
+        2. Create resource(s) filter(s) without overriding them
+        3. Create user with taxonomies same as role taxonomies
+        4. Assign step 1 role to user
+
+        @assert: User should be able to access the resource(s) of the assigned
+        role
+
+        @caseautomation: notautomated
+
+        @CaseLevel: System
```

Comment:
@CaseLevel: system is valid for tier3  not tier1
and for tier2, @caselevel should be Integration.